api/python/provisioner/inputs.py

```diff
@@ -1002,7 +1002,8 @@ class SWUpgradeRepo(SWUpdateRepo):
                 'choices': list(map(lambda elem: elem.value, config.HashType))
             }
         },
-        validator=attr.validators.optional(attr.validators.instance_of(str)),
+        validator=attr.validators.optional(
+            attr.validators.in_(config.HashType)),
```

Comment:
Just a suggestion. 
It helps if the closing ')' is on an independent line.

Reply:
It is a matter of taste. PEP8 allows both ways. I love the compact approach. But I fixed it here.